### ISSUE TITLE
Add an option to specify minimum required environment versions

### ISSUE THREAD

**Issue by Lehoczky** [es-tooling/module-replacements#16]
It would be very helpful if we had the ability to specify a minimum Node.js or ECMAScript version for a recommendation.

For example, [is-builtin-module](https://github.com/es-tooling/module-replacements/blob/main/docs/modules/is-builtin-module.md) can only be replaced for node versions above v16.

**Comment by 43081j** [comment]
you're right, although the documentation for that one could be improved too

`builtinModules` is available since 8.x or something, so we should actually document that users on `<16` can do the following:

```ts
import {builtinModules} from 'node:module';

builtinModules.includes(specifier.replace(/^node:/, ''));
```
